### DISCUSSION TITLE
Having problems getting webhook to work

### DISCUSSION THREAD

**Discussion by Grympel** [mbround18/valheim-docker#253] [Help!] [answered]
I'm most likely doing something stupid or wrong, but I simply can't get the webhook to work.

**My configuration using portainer:**
version: "3"
services:
  valheim:
    image: mbround18/valheim:latest
    ports:
      - 2456:2456/udp
      - 2457:2457/udp
      - 2458:2458/udp
    environment:
      - PORT=2456
      - NAME="Servername"
      - WORLD="Dedicated"
      - PASSWORD="Password"
      - TZ=Europe/Oslo
      - PUBLIC=1
      - AUTO_UPDATE=1
      - AUTO_UPDATE_SCHEDULE="0 1 * * *"
      - AUTO_BACKUP=1
      - AUTO_BACKUP_SCHEDULE="*/15 * * * *"
      - AUTO_BACKUP_REMOVE_OLD=1
      - AUTO_BACKUP_DAYS_TO_LIVE=3
      - AUTO_BACKUP_ON_UPDATE=1
      - AUTO_BACKUP_ON_SHUTDOWN=1
      - WEBHOOK_URL="https://discordapp.com/api/webhooks/something/something"
      - WEBHOOK_UPDATING_MESSAGE="Updating, please wait"
      - WEBHOOK_UPDATE_SUCCESSFUL_MESSAGE="Update complete!"
      - WEBHOOK_START_SUCCESSFUL_MESSAGE="Server up and running!"
      - WEBHOOK_STOPPING_MESSAGE="Server going down"
      - UPDATE_ON_STARTUP=1
    volumes:
      - ./valheim/saves:/home/steam/.config/unity3d/IronGate/Valheim
      - ./valheim/server:/home/steam/valheim
      - ./valheim/backups:/home/steam/backups
      
**Output from ODIN:**
root@077f2fcfdf51:/home/steam/steamcmd# odin -d notify test
[ODIN][DEBUG] - Debugging set to true
[ODIN][DEBUG] - Debug mode enabled!
[ODIN][DEBUG] - Launching notify command...
[ODIN][DEBUG] - Checking env for MESSAGE
[ODIN][DEBUG] - Checking env for webhook_url
[ODIN][DEBUG] - Env variable found webhook_url="https://discordapp.com/api/webhooks/something/something"
[ODIN][INFO]  - Sending Broadcast: test
[ODIN][DEBUG] - Webhook enabled, sending notification Broadcast
[ODIN][DEBUG] - Webhook URL: "https://discordapp.com/api/webhooks/something/something"
[ODIN][DEBUG] - Webhook Payload: {"event_type":{"name":"Broadcast","status":"Triggered"},"event_message":"test","timestamp":"2021-03-04T19:53:08.145458290+01:00"}
[ODIN][ERROR] - [Broadcast]: Error with webhook! Status 500
root@077f2fcfdf51:/home/steam/steamcmd#

**Comment by CosmicHorrorDev** [comment, marked as answer]
That payload doesn't look right for discord. A fix for the discordapp base was added a few days ago. Are you on an older image?

**Reply by Grympel** [reply]
Rebuilt the entire thing about 30 mins ago

**Reply by CosmicHorrorDev** [reply]
So just to check when you run `docker image ls --digest` the digest listed for `mbround18/valheim` is `sha256:ba15683bfb677452fa31496488c76a9c74511bb9eb7b39c4b801a426d6e281b2`?

I don't know how to get the discordapp webhook url vs just discord, so I can't test locally, but yours should be picked up correctly ([playground demo](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=42ac3d951b44767aba3d9b60f22d6bca))

And just to note, the custom webhook messages set with `WEBHOOK_UPDATING_MESSAGE` and friends, won't work all the time right now, but that should be fixed later today.

**Reply by Grympel** [reply]
I'm getting  sha256:5dfadc1aade32baf37eb2fa76627800c32396eb9925943ded2c5c1ef4fd33114
must have managed to pull the wrong one somehow

Was using the messeges to test :)

**Reply by Grympel** [reply]
I've pulled every image I could get my hands on and none matches the sha256 you linked. No idea what I'm doing wrong not to find it.

sha256:5dfadc1aade32baf37eb2fa76627800c32396eb9925943ded2c5c1ef4fd33114 mbround18/valheim:latest
sha256:36f01942ef5ad665f5a3c0e7fc9a35f6f74d285f6f9113ee8889624213769345 mbround18/valheim:development-status
sha256:b9aa682584b0080118aa833b02ba3b87955abc5ef7d82f8b22e8d2d58afa3dba mbround18/valheim:development-webhook
sha256:56d3e7a691e8589300cfc840c6e7bc6614b56e72bde62c1a73123939911ec644 mbround18/valheim:development-bepinex
sha256:387198fc744a7ded2bee9878c1e6c436f8c324a2bf5df019c3630d38e9d67e23 mbround18/valheim:1.2.0
sha256:c49ed1073a78f069a5bf4819eb8f1b5dd033d39e27293d848130c3a635ae35ff mbround18/valheim:1.1.1
sha256:05970858f741f9093087fa57707e65108edc44623c168919ebf3bc32aecac246 mbround18/valheim:1.1.0
sha256:eb13bbb4ab5dd9d5a39c38a3d15719dd429435f729d43bd1f972bbebfcb8de3e mbround18/valheim:1.0.0

**Reply by CosmicHorrorDev** [reply]
Your compose file should not have double quotes within the string for WEBHOOK_URL. It should be `"WEBHOOK_URL=..."` if you're going to have double quotes

**Reply by mbround18** [reply]
pull the latest image as well

**Reply by Grympel** [reply]
Woke up in the middle of the night from a stop/stop/start so the cron part worked. Did another pull witohut the double quotes and got my "Server up and running!", also with double quotes :)

Thanks for the help! I did something stupid after all :P Maybe that's another feature to add? Stripping of double quotes if it's possible to have less people like me asking for help :)

PS! You're using double quotes in the "Everything but the kitchen sink" example.

**Reply by CosmicHorrorDev** [reply]
I think fixing the example would be better than stripping double quotes in case people want double quotes in their server name or password.

Would definitely be a good thing to fix the example. There's another style for the docket yaml syntax that makes the problem less likely to happen!
